README.md

```diff
@@ -184,7 +184,8 @@ Other potentially useful environment variables may be found in `setup.py`.
 **Common**
 
 ```bash
-conda install astunparse numpy ninja pyyaml setuptools cmake cffi typing_extensions future six requests dataclasses
+conda install cmake ninja
+pip install -r requirements.txt
 ```
```

Comment:
### Dropped dependencies
```
cffi              (inlined) # Did not affect build even though it's suggested here https://github.com/pytorch/pytorch/blob/ae0c4c4c295b1b4d6ac94f3c665bc39c7db052cd/.github/requirements/conda-env-macOS-ARM64#L5
dataclasses       (inlined) # We assume python 3.7+ so shouldn't need https://github.com/pytorch/pytorch/pull/78501 also "#this breaks some platforms circleci.txt"
```
### Added from requirements.txt:
```
types-dataclasses (requirements.txt)
filelock          (requirements.txt) # Required for testing see torch/_inductor/codecache.py
psutil            (requirements.txt, requirements-ci.txt) # Required in .circleci/docker/requirements-ci.txt
expecttest        (requirements.txt, requirements-ci.txt) # Required for unit testing, see contributing doc. https://github.com/pytorch/pytorch/pull/63320
hypothesis        (requirements.txt, requirements-ci.txt) # Required for unit testing, see contributing doc, torch/testing/_internal/common_utils.py inductor/test_perf.py
```

### Unchanged
```
cmake             Required for build on macos
ninja             Required for build on macos
jinja2            (requirements.txt, setup.py  )
sympy             (requirements.txt, setup.py  ) 
networkx          (requirements.txt, setup.py, requirements-ci.txt)
setuptools        (requirements.txt, setup.py, inlined)
typing_extensions (requirements.txt, setup.py, inlined)
astunparse        (requirements.txt, inlined)
future            (requirements.txt, inlined)
numpy             (requirements.txt, inlined)
pyyaml            (requirements.txt, inlined)
requests          (requirements.txt, inlined)
six               (requirements.txt, inlined)
```

# Would make sense to add, but out of scope of PR
```
pytest            (requirements-ci.txt)# Expected by distributions/test_constraints.py see https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#better-local-unit-tests-with-pytest and USE_PYTEST_LIST = [ .circleci/docker/requirements-ci.txt
click             (requirements-ci.txt)# RuntimeError: Failed to import torch.csrc.jit.tensorexpr.scripts.bisect: No module named 'click' + .circleci/docker/requirements-ci.txt
xdoctest          (requirements-ci.txt)# test/run_test.py", line 668, in run_doctests  + .circleci/docker/requirements-ci.txt
```


<details>
  <summary>(Additional deps recommended from requirements-ci.txt)</summary>
```
boto3
coremltools
flatbuffers
junitparser
librosa
mypy
numba   
opt-einsum
protobuf
pygments
pytest
pytest-flakefinder==1.1.0
pytest-rerunfailures
pytest-shard
pytest-xdist
scikit-image
scipy
tb-nightly
unittest-xml-reporting
```
</details>

Reply:
Just curious, why not install `cmake` and `ninja` from requirements.txt as well?

Reply:
Good question, so `ninja` and `cmake` are as far as I understand not python dependencies so putting them in `requirements.txt` would probably work with `conda install --file requirementst.txt` but it would be confusing for everyone trying to `pip install`

Reply:
- [cmake-on-pypi](https://pypi.org/project/cmake/)
- [ninja-on-pypi](https://pypi.org/project/ninja/)

Reply:
Good catch! I'm not familiar with what any of these packages are, there seems to be various instructions on how to install.

This one is  addressed in https://github.com/pytorch/pytorch/pull/91850 where I suggest we should point to install instruction in readme fixed in this PR:

https://github.com/pytorch/pytorch/blob/4d26903739fa3857755f5508b7c9dba39c9c53fa/CONTRIBUTING.md?plain=1#L89

Others, I do not know anything about:
https://github.com/pytorch/pytorch/blob/c371542efc31b1abfe6f388042aa3ab0cef935f2/scripts/README.md?plain=1#L32

https://github.com/pytorch/pytorch/blob/daff5d35567615bb80f19e59474d8af7af84daf2/caffe2/release-notes.md?plain=1#L14-L16